processor/metricstransformprocessor/README.md

```diff
@@ -0,0 +1,97 @@
+- Rename label values (e.g. rename `done` to `complete`)
+- Aggregate across label sets (e.g. only want the label `usage`, but don’t care about the labels `core`, and `cpu`)
+  - Aggregation_type: sum, average, max
+- Aggregate across label values (e.g. want `memory{slab}`, but don’t care about `memory{slab_reclaimable}` & `memory{slab_unreclaimable}`)
+  - Aggregation_type: sum, average, max
+
+## Configuration
+```yaml
+# name is used to match with the metric to operate on. This implementation doesn’t utilize the filtermetric’s MatchProperties struct because it doesn’t match well with what I need at this phase. All is needed for this processor at this stage is a single name string that can be used to match with selected metrics. The list of metric names and the match type in the filtermetric’s MatchProperties struct are unnecessary. Also, based on the issue about improving filtering configuration, it seems like this struct is subject to be slightly modified.
+name: <current_metric_name>
+
+# action specifies if the operations are performed on the current copy of the metric or on a newly created metric that will be inserted
+action: {update, insert}
+
+# new_name is used to rename metrics (e.g. rename cpu/usage to cpu/usage_time) if action is insert, new_name is required
+new_name: <new_metric_name_inserted>
+
+# operations contain a list of operations that will be performed on the selected metrics. Each operation block is a key-value pair, where the key can be any arbitrary string set by the users for readability, and the value is a struct with fields required for operations. The action field is important for the processor to identify exactly which operation to perform 
+operations:
+
+  # update_label action can be used to update the name of a label or the values of this label (e.g. rename label `cpu` to `core`)
+  -action: update_label
+   label: <current_label1>
+   new_label: <new_label>
+   value_actions:
+     -value: <current_label_value>
+      new_value: <new_label_value>
+
+  # aggregate_labels action aggregates metrics across labels (e.g. only want the label `usage`, but don’t care about the labels `core`, and `cpu`)
+  -action: aggregate_labels
+   # label_set contains a list of labels that will remain after the aggregation. The excluded labels will be aggregated by the way specified by aggregation_type.
+   label_set: [labels...]
+   aggregation_type: {sum, average, max}
+
+  # aggregate_label_values action aggregates labels across label values (e.g. want memory{slab}, but don’t care about memory{slab_reclaimable} & memory{slab_unreclaimable})
+  -action: aggregate_label_values
+   label: <label>
+   # aggregated_values contains a list of label values that will be aggregated by the way specified by aggregation_type into new_value. The excluded label values will remain.
+   aggregated_values: [values...]
+   new_value: <new_value> 
+   aggregation_type: {sum, average, max}
+```
+
+## Examples
+
+### Insert New Metric
```

Comment:
What would the config look like if I'd like to rename more than one metric? I maybe missing the context, but it appears to me like one can only operate on a single metric per processor config. Is that correct?

Reply:
Sorry, it was actually a bit out of date. I just updated the new version, which has a `transforms` field that accepts a list, which can be used to transform more than one metric.

Reply:
Now that your other PR is quite far along, I recommend you just incorporate this README into your other PR and close this one. We can merge this together with the implementation once that's ready

Reply:
Yea that works too!